backend/pvm.pm

```diff
@@ -0,0 +1,164 @@
+package backend::pvm;
+use strict;
+use base ('backend::baseclass');
+use bmwqemu qw(fileContent diag save_vars);
+require IPC::System::Simple;
+use autodie qw(:all);
+use File::Basename;
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new;
+
+    $self->{pid}         = undef;
+    $self->{children}    = [];
+    $self->{pidfilename} = 'pvm.pid';
+    $self->{pvmctl}      = '/usr/bin/pvmctl';
+    die "pvmctl not found" unless -x $self->{pvmctl};
+
+    return $self;
+}
+
+sub do_start_vm {
+    my $self = shift;
+    $self->unlink_crash_file;
+    $self->start_lpar();
+    return {};
+}
+
+sub runcmd {
+    diag "running " . join(' ', @_);
+    return system(@_);
+}
+
+sub pvmctl {
+    my $self = shift @_;
+    my $vars = \%bmwqemu::vars;
+    my ($type, $action) = @_;
+    push(my @cmd, $self->{pvmctl}, $type, $action);
+    my $lpar = $vars->{LPAR};
+
+    if ($type =~ /lpar/) {
+        push(@cmd, "-i", "name=" . $lpar) if ($action =~ /power|restart|delete/);
+        if ($action =~ /create/) {
+            my ($type, $action, $cpu, $memory) = @_;
+            push(@cmd, qw{--proc-type shared --sharing-mode uncapped --type AIX/Linux});
+            push(@cmd, "--proc", $cpu, "--mem", $memory, "--name", $lpar);
+            push(@cmd, "--proc-unit", $cpu * 0.05);
+        }
+    }
+    elsif ($type =~ /scsi/) {
+        my ($type, $action, $kind, $file) = @_;
+        push(@cmd, "--type", $kind, "--stor-id", $file);
+        push(@cmd, "--lpar", "name=" . $lpar);
+        push(@cmd, "--vg", "name=rootvg") if ($type =~ /lv/);
+    }
+    elsif ($type =~ /lv/) {
+        my ($type, $action, $name, $size) = @_;
+        push(@cmd, "--name", $name, "--size", $size);
+    }
+    elsif ($type =~ /eth/) {
+        my ($type, $action, $vlan, $vswitch) = @_;
+        push(@cmd, "--pvid", $vlan, "--vswitch", $vswitch, "-p", "name=" . $lpar);
+    }
+    else {
+        die "Unrecognized command $type";
+    }
+    runcmd(@cmd);
+}
+sub attach_console {
+    my $vars    = \%bmwqemu::vars;
+    my $vncport = qx{sudo /usr/sbin/mkvtermutil --id $vars->{LPARID} --vnc --local --log serial0 2>/dev/null};
+    $vncport =~ /([0-9]+)/;
+    chomp($vncport);
+    $vars->{VNC} = $vncport;
+    diag "VNC is $vars->{VNC}";
+}
+
+sub start_lpar {
+    my $self = shift;
+    my $vars = \%bmwqemu::vars;
+    #general settiings
+    $vars->{LPAR} = "osauto" . $vars->{WORKER_ID};
+    $vars->{CPUS} ||= 1;
+    $vars->{MEM}  ||= "2048";
+    #disk settings
+    $vars->{NUMDISKS}  ||= 1;
+    $vars->{HDDSIZEGB} ||= 15;
+    #network settings
+    $vars->{NIC}     ||= "sea";
+    $vars->{NICVLAN} ||= 1;
+    $vars->{VSWITCH} ||= "ETHERNET0";
+    # generate the ISO name for VIO
+    my $os      = $vars->{DISTRI};
```

Comment:
Yes. But this limitation is really stupid and I'm afraid I can't make it fixed. (Those parts are closed source). And I want to keep similarity across architectures, The iso name just changed temporarily in background.